geoguide/client/static/src/pageEnvironment/index.js

```diff
@@ -35,6 +35,59 @@ let currentChartIndex = 0
 let mouseClusters = {}
 let mousePolygons = []
 
+if (typeof(Storage) !== "undefined") {
+  let filename = JSON.parse(document.querySelector('#dataset_json').innerHTML).filename;
+  let ls;
+  if(localStorage.getItem(filename) == null) {
+    ls = {
+      "colorModifier": document.querySelector('#colorModifier').selectedIndex,
+      "sizeModifier": document.querySelector('#sizeModifier').selectedIndex,
+      "timelimit": document.getElementById("timelimit").value,
+      "sigma": document.getElementById("sigma").value,
+      "kvalue": document.getElementById("kvalue").value,
+      "onlyfilteredpoints": document.getElementById("onlyfilteredpoints").getAttribute('checked')
+    };
+    
+  } else {
+     ls = JSON.parse(localStorage.getItem(filename));
+
+     document.querySelector('#colorModifier').selectedIndex = ls["colorModifier"];
+     document.querySelector('#sizeModifier').selectedIndex = ls["sizeModifier"];
+     document.getElementById("timelimit").value = ls["timelimit"];
+     document.getElementById("sigma").value = ls["sigma"];
+     document.getElementById("kvalue").value = ls["kvalue"];
+     document.getElementById("onlyfilteredpoints").setAttribute("checked", ls['onlyfilteredpoints']);
+  }
+
+  document.querySelector('#colorModifier').addEventListener('change', e => {
```

Comment:
We don't need to listen to these changes... On `onbeforeunload` callback, we can get the values from DOM before save.